### DISCUSSION TITLE
Errordef value when using iMinuit as minimiser

### DISCUSSION THREAD

**Discussion by reallyblaised** [scikit-hep/pyhf#2517] [Q&A] [answered]
Hello experts,

I just wondered if in a scenario of a binned maximum likelihood fit with `pyhf` and `iMinuit` as minimiser, I must manually modify the `errordef` to counteract potential factors of 2, or whether this is automatically taken care of in defining the likelihood internally. 

Pragmatically, I was wondering whether I should use this snippet at the start of my fitting code:

```python
  pyhf.set_backend(
    # https://scikit-hep.org/pyhf/_generated/pyhf.optimize.opt_minuit.minuit_optimizer.html
    # 0.5 errordef for NLL
    "numpy",
    pyhf.optimize.minuit_optimizer(verbose=1, errordef=0.5),
    )  # get uncertainties and covariance
    pyhf_fit_result, result_obj = pyhf.infer.mle.fit(
        data, pdf, return_result_obj=True, return_uncertainties=True
    )
```

to extract the correct error estimate. I have seen related differences in other discussions, but I thought it useful to confirm this once and for all. 

Thanks a lot in advance!

**Comment by alexander-held** [comment, marked as answer]
Hi, this relates back to #1006. Currently `pyhf` uses 2*NLL as objective function to minimize: https://github.com/scikit-hep/pyhf/blob/e8789a2a5233f1c4d825ef0daeec8434f6893146/src/pyhf/infer/mle.py#L132-L134

With this definition, `errordef=1` is the correct thing to use, which is what `pyhf` does internally as well: https://github.com/scikit-hep/pyhf/blob/e8789a2a5233f1c4d825ef0daeec8434f6893146/src/pyhf/optimize/opt_minuit.py#L41

There is no need to manually change `errordef`. If `pyhf` changes to the NLL as objective, the `errordef` would have to also be changed at the same time. If you run the fit externally, the right value depends on what you use as objective. You can do 0.5 and NLL or 1.0 and 2*NLL (or in principle any other multiple of those).
